File: enode/src/main/java/org/enodeframework/domain/IAggregateRootFactory.java
package org.enodeframework.domain;

/**
 * Defines a factory to create empty aggregate root.
 */
public interface IAggregateRootFactory {
    /**
     * Create an empty aggregate root with the given type.
     */
    <T extends IAggregateRoot> T createAggregateRoot(Class<T> aggregateRootType);
}

File: enode/src/main/java/org/enodeframework/domain/IAggregateRootFactory.kt
package org.enodeframework.domain

/**
 * Defines a factory to create empty aggregate root.
 */
interface IAggregateRootFactory {
    /**
     * Create an empty aggregate root with the given type.
     */
    fun <T : IAggregateRoot?> createAggregateRoot(aggregateRootType: Class<T>): T
}